### ISSUE TITLE
./x.py setup using option "d" (maintainer) is broken in 1.48.0

### ISSUE THREAD

**Issue by maruel** [rust-lang/rust#79933]
### Code

Following https://rustc-dev-guide.rust-lang.org/getting-started.html

```
git clone --recurse -b 1.48.0 https://github.com/rust-lang/rust.git
cd rust
./x.py setup
# Choose "d) Install Rust from source"
./x.py check
```

I expected to see this happen: works

Instead, this happened: 

```
$ cat config.toml
# Includes one of the default files in src/bootstrap/defaults
profile = "maintainer"

$ ./x.py check
Updating only changed submodules
Submodules updated in 0.06 seconds
    Finished dev [unoptimized + debuginfo] target(s) in 0.13s
thread 'main' panicked at 'fs::read_to_string(file) failed with No such file or directory (os error 2) ("`include` config not found")', src/bootstrap/config.rs:563:28
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
failed to run: /usr/local/google/home/maruel/src-oth/rust/build/bootstrap/debug/bootstrap check
Build completed unsuccessfully in 0:00:00

$ ls src/bootstrap/defaults
config.toml.codegen  config.toml.compiler  config.toml.library  config.toml.user  README.md
```

It's because src/bootstrap/defaults/ doesn't contain config.toml.maintainer

### Version it worked on

1.47.0 doesn't have `x.py setup`.

### Version with regression

1.48.0

**Comment by maruel** [comment]
@jyn514 created the directory.

**Comment by maruel** [comment]
The fix seems to be

```
git move src/bootstrap/defaults/config.toml.user to src/bootstrap/defaults/config.toml.maintainer.
```

Workaround:

```
touch src/bootstrap/defaults/config.toml.maintainer
```

as in my case, I wanted to build from scratch, so config.toml.user were not defaults I wanted to use.

**Comment by jyn514** [comment]
This does work on master, I think it was fixed in https://github.com/rust-lang/rust/pull/77400 but never backported.

@Mark-Simulacrum what's the policy for source-only backports? Backporting https://github.com/rust-lang/rust/pull/77400 to 1.48 wouldn't need a point release, just a cherry-pick on the stable branch.

**Comment by Mark-Simulacrum** [comment]
We do not usually land things onto the stable branch unless we are intending to do a point release (though there's no particular reason why, beyond the fact that the branch is not really meaningful since people should be using the tags for stable checkouts).

**Comment by maruel** [comment]
@jyn514 sorry I guess I shouldn't have put my workaround in my comment. I explicitly wanted to build from scratch, and config.toml.user doesn't do that. That's irrelevant to the issue here.

re point release: I'm fine with a WontFix here since it was fixed on tip of tree.

**Comment by jyn514** [comment]
@maruel config.toml.user *does* do that, I'm confused why you think it doesn't.

**Comment by jyn514** [comment]
I'm not trying to berate you, just see how I can improve the defaults in .user if they don't match your needs. I think you're the first person to try to use it.

**Comment by maruel** [comment]
You are right, I misread https://github.com/rust-lang/rust/blob/master/src/bootstrap/defaults/config.user.toml.

**Comment by jyn514** [comment]
Since this is fixed in 1.49 and there's no plan to backport a fix to 1.48, I'm going to close this. Feel free to give feedback about the 'maintainer' defaults, but I think belongs in a separate issue.
